doc/esmvalcore/preprocessor.rst

```diff
@@ -138,344 +260,559 @@ losing generality. To mask ice out, `mask_landseaice` can be used:
+  season.
+* ``extract_month``: Extract only the times that occur within a specific month.
+* ``time_average``: Take the weighted average over the time dimension.
+* ``seasonal_mean``: Produces a mean for each season (DJF, MAM, JJA, SON)
+* ``annual_mean``: Produces an annual or decadal mean.
+* ``regrid_time``: Aligns the time axis of each dataset to have common time
+  points and calendars.
+
+``extract_time``
+----------------
 
 This function subsets a dataset between two points in times. It removes all
 times in the dataset before the first time and after the last time point.
 The required arguments are relatively self explanatory:
 
-* start_year
-* start_month
-* start_day
-* end_year
-* end_month
-* end_day
+* ``start_year``
+* ``start_month``
+* ``start_day``
+* ``end_year``
+* ``end_month``
+* ``end_day``
 
-These start and end points are set using the datasets native calendar. All six
-arguments should be given as integers, named month strings (e.g., March) will
-not be accepted. Note that start_year and end_year can be omitted, as they are
-filled in automatically from the dataset definition if not specified
-here (end_year will be the value in the dataset definition + 1).
+These start and end points are set using the datasets native calendar.
+All six arguments should be given as integers - the named month string
+will not be accepted.
 
 See also :func:`esmvalcore.preprocessor.extract_time`.
 
-2. extract_season
------------------
+``extract_season``
+------------------
 
 Extract only the times that occur within a specific season.
 
-This function only has one argument:
-
-* season: DJF, MAM, JJA, or SON
+This function only has one argument: ``season``. This is the named season to
+extract. ie: DJF, MAM, JJA, SON.
 
 Note that this function does not change the time resolution. If your original
 data is in monthly time resolution, then this function will return three
 monthly datapoints per year.
 
-To calculate a seasonal average, this function needs to be combined with the
-seasonal_mean function, below.
+If you want the seasonal average, then this function needs to be combined with
+the seasonal_mean function, below.
 
 See also :func:`esmvalcore.preprocessor.extract_season`.
 
-3. extract_month
-----------------
+``extract_month``
+-----------------
 
 The function extracts the times that occur within a specific month.
-This function only has one argument:
-
-* month: [1-12]
-
-Note that named month strings will not be accepted.
+This function only has one argument: ``month``. This value should be an integer
+between 1 and 12 as the named month string will not be accepted.
 
 See also :func:`esmvalcore.preprocessor.extract_month`.
 
-4. time_average
----------------
+.. _time_average:
+
+``time_average``
+----------------
 
-This function takes the weighted average over the time dimension. This
+This functions takes the weighted average over the time dimension. This
```

Comment:
Should be "function" instead of "functions"?

Reply:
Yes